bin/tictactoe

```diff
@@ -1,3 +1,42 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+require_relative '../lib/game.rb'
+require_relative '../lib/players/computer.rb'
+require_relative '../lib/players/human.rb'
+
+puts "Greetings! Welcome to Tic Tac Toe. "
+
+continue_play = true
+
+while continue_play do
+
+  puts "How many players?"
+  player_number = gets.chomp
+
+    if player_number.to_i == 2
+      game = Game.new
+      game.play
+    elsif player_number.to_i == 1
+      puts "Would you like to start first? Y/N"
+      if gets.chomp!="Y"
+        game = Game.new(Computer.new("X"), Human.new("O"))
+        game.play
+      else
+        game = Game.new(Human.new("X"), Computer.new("O"))
+        game.play
+      end
+    elsif player_number.to_i == 0
+      game = Game.new(Computer.new("X"), Computer.new("O"))
+      game.play
+    else
+      puts "please enter 0, 1 or 2."
+    end
+
+    puts "Would you like to play another game? Y/N"
+    answer = gets.chomp
+    if answer != "Y"
+      continue_play = false
+    end
+
+end
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.